### PR TITLE
Publish Feb 4th, 2021 - routerLink

### PR DESCRIPTION
**Pull request by afebbraro**
## What does this PR do?
Updates components to be able to use routerLinks.

### Documentation
 - [x] Update Spark Docs - updated some SB docs

### Code
 - [x] Build Component in Angular
 - [x] Unit Testing in HTML with `npm run test` in `html/` (100% coverage, 100% passing)
 - [x] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)
 - [x] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### CODE REVIEW

**Review by guern1kn — APPROVED** [review]

File: angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.spec.ts [hunk 1]
```diff
@@ -157,45 +160,97 @@ describe('SprkDropdownComponent', () => {
     expect(dropDownComponent.choices[0]['active']).toEqual(false);
   });
 
+  it('should href value if href is set on choice item', () => {
```
Comment (bobwise) [inline angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.spec.ts:163]:
```suggestion
  it('should set href value if href is set on choice item', () => {
```

File: angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.spec.ts [hunk 2]
```diff
@@ -157,45 +160,97 @@ describe('SprkDropdownComponent', () => {
...
+    const listLink = fixture.nativeElement.querySelector(
+      '.sprk-c-Dropdown__link',
+    );
+    listElement.dispatchEvent(new Event('click'));
+    fixture.detectChanges();
+    expect(dropDownComponent.choices[0]['active']).toEqual(false);
+    expect(listLink.getAttribute('href')).toEqual('/test');
+  });
+
+  it('should href value if routerLink is set on choice item', () => {
```
Comment (bobwise) [inline angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.spec.ts:189]:
```suggestion
  it('should set href value if routerLink is set on choice item', () => {
```

File: angular/projects/spark-angular/src/lib/components/sprk-footer/sprk-footer.component.spec.ts [hunk 3]
```diff
@@ -40,7 +42,7 @@ describe('SprkFooterComponent', () => {
     component.additionalClasses = 'sprk-u-pam sprk-u-man';
     fixture.detectChanges();
     expect(component.getClasses()).toEqual(
-      'sprk-o-CenteredColumn sprk-o-Stack sprk-o-Stack--misc-b sprk-c-Footer sprk-u-pam sprk-u-man'
+      'sprk-o-CenteredColumn sprk-o-Stack sprk-o-Stack--misc-b sprk-c-Footer sprk-u-pam sprk-u-man',
```
Comment (bobwise) [inline angular/projects/spark-angular/src/lib/components/sprk-footer/sprk-footer.component.spec.ts:45]: I think we should add some new tests for the new routerLink options.

File: angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.component.spec.ts [hunk 4]
```diff
@@ -198,17 +213,47 @@ describe('SprkMastheadComponent', () => {
...
+
+  it('should set the href link of the bigNavLink when using routerLink', () => {
+    component.bigNavLinks = [{ text: 'Item 1', routerLink: '/alert-router' }];
+    fixture.detectChanges();
+    const secondaryNavElementItem = fixture.nativeElement.querySelector(
+      '.sprk-c-Masthead__link--big-nav',
+    );
+    expect(secondaryNavElementItem.getAttribute('href')).toEqual(
+      '/alert-router',
+    );
```
Comment (bobwise) [inline angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.component.spec.ts:257]: Tests for more routerLink options. Do we need a test for logo-slot/logo-slot-router?

**Review by bobwise — CHANGES_REQUESTED** [review]
Great job on this! Left a few comments.

File: angular/projects/spark-angular/src/lib/components/sprk-award/sprk-award.component.spec.ts [hunk 5]
```diff
@@ -288,6 +343,7 @@ describe('SprkAwardComponent', () => {
   it('should prefer imgTwoAdditionalClasses over additionalClassesImgTwo', () => {
     const testOldProp = 'testClass';
     const testNewProp = 'testNewClass';
+    component.imgTwoHref = '/test';
```
[the remark this answers is not in the document]
Comment (afebbraro) [reply angular/projects/spark-angular/src/lib/components/sprk-award/sprk-award.component.spec.ts:346]: In the component file, if imgTwoHref is not set, then there is no image. This fixes an old test.

**Review by bobwise — APPROVED** [review]

**Pull request merged**